packages/zowe-explorer-api/src/profiles/ZoweExplorerApi.ts

```diff
@@ -147,7 +124,7 @@ export namespace ZoweExplorerApi {
          * @param {zowe.IUploadOptions} [options]
          * @returns {Promise<zowe.IZosFilesResponse>}
          */
-        putContent?(inputFilePath: string, ussFilePath: string, options?: zowe.IUploadOptions): Promise<zowe.IZosFilesResponse>;
```

Comment:
I removed the option here to reflect what we had for other.  I am wondering if we should make all APIs in this file an option vs a requirement so more extenders could make use of only what they need.  What do y'all think?

Reply:
If we do, we may need a "standard" way of handling when a method has not been implemented by an extender.

Maybe a small utility function that take a function name (as a string), validates it against the corresponding interface (which we may need to convert to `type`s to do so), and shows an pop-up if it doesn't exist with a standard message or calls it if it exists.

Reply:
that sounds like a good plan but maybe out of scope for this PR.  We could discuss further if we would like that change for v3